romeo/src/state.rs

```diff
@@ -0,0 +1,218 @@
+/// It returns any new I/O tasks the system need to perform alongside the updated state.
+#[tracing::instrument(skip(config, state))]
+pub fn update(config: &Config, state: State, event: Event) -> (State, Vec<Task>) {
+    debug!("Handling update");
+
+    match event {
+        Event::StacksTransactionUpdate(txid, status) => {
+            process_stacks_transaction_update(config, state, txid, status)
+        }
+        Event::BitcoinTransactionUpdate(txid, status) => {
+            process_bitcoin_transaction_update(config, state, txid, status)
+        }
+        Event::BitcoinBlock(block) => process_bitcoin_block(config, state, block),
+        Event::AssetContractCreated(txid) => process_asset_contract_created(config, state, txid),
+        event => panic!("Cannot handle yet: {:?}", event),
+    }
+}
+
+fn process_bitcoin_block(_config: &Config, mut state: State, block: Block) -> (State, Vec<Task>) {
+    let deposits = parse_deposits(&block);
+    let withdrawals = parse_withdrawals(&block);
+
+    state.deposits.extend_from_slice(&deposits);
+    state.withdrawals.extend_from_slice(&withdrawals);
+
+    state.block_height = block
+        .bip34_block_height()
+        .expect("Failed to get block height");
+
+    let mut tasks = vec![Task::FetchBitcoinBlock(state.block_height + 1)];
+
+    let mint_tasks = state
+        .deposits
+        .iter_mut()
+        .filter_map(|deposit| deposit.mint(state.block_height));
+
+    let withdrawals = &mut state.withdrawals;
+
+    let burn_tasks: Vec<_> = withdrawals
+        .iter_mut()
+        .filter_map(|withdrawal| withdrawal.burn(state.block_height))
+        .collect();
+
+    let fulfillment_tasks: Vec<_> = withdrawals
+        .iter_mut()
+        .filter_map(|withdrawal| withdrawal.fulfill(state.block_height))
+        .collect();
+
+    tasks.extend(mint_tasks);
+    tasks.extend(burn_tasks);
+    tasks.extend(fulfillment_tasks);
+
+    (state, tasks)
+}
+
+fn parse_deposits(_block: &Block) -> Vec<Deposit> {
+    // TODO: #67
+    vec![]
+}
+
+fn parse_withdrawals(_block: &Block) -> Vec<Withdrawal> {
+    // TODO: #68
+    vec![]
+}
+
+fn process_bitcoin_transaction_update(
+    _config: &Config,
+    state: State,
+    _txid: BitcoinTxId,
+    _status: TransactionStatus,
+) -> (State, Vec<Task>) {
+    // TODO: #67 and #68
+
+    (state, vec![])
+}
+
+fn process_stacks_transaction_update(
+    _config: &Config,
+    state: State,
+    _txid: StacksTxId,
+    _status: TransactionStatus,
+) -> (State, Vec<Task>) {
+    // TODO: #73 and #68
+    (state, vec![])
+}
+
+fn process_asset_contract_created(
+    _config: &Config,
+    mut state: State,
+    txid: StacksTxId,
+) -> (State, Vec<Task>) {
+    // TODO: #73
+    state.contract = Some(Contract {
+        txid,
+        status: TransactionStatus::Broadcasted,
+    });
+
+    let task = Task::CheckStacksTransactionStatus(txid);
+
+    (state, vec![task])
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Biased here but wouldn't it be best to separate test stubs into a different PR? if you merge them as is, with CI failing. Any subsequent PR will need to bypass CI or lockstep the next PR to implement the tests.

Reply:
Oh right yeah let's get rid of the stubs.

Reply:
Done :heavy_check_mark: